objects/src/notes/location.rs

```diff
@@ -1,34 +1,37 @@
-use alloc::string::ToString;
-
 use super::{
-    ByteReader, ByteWriter, Deserializable, DeserializationError, Digest, NoteError, Serializable,
-    NOTE_TREE_DEPTH,
+    ByteReader, ByteWriter, Deserializable, DeserializationError, NoteError, Serializable,
 };
-use crate::crypto::merkle::{MerklePath, NodeIndex};
+use crate::{crypto::merkle::MerklePath, MAX_BATCHES_PER_BLOCK, MAX_OUTPUT_NOTES_PER_BATCH};
 
-/// Contains information about the origin of a note.
+/// Contains information about the location of a note.
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct NoteOrigin {
+pub struct NoteLocation {
     /// The block number the note was created in.
-    pub block_num: u32,
+    block_num: u32,
 
     /// The index of the note in the note Merkle tree of the block the note was created in.
-    pub node_index: NodeIndex, // TODO: should be a u32 because the depth is always the same
+    node_index_in_block: u32,
+}
+
+impl NoteLocation {
+    /// Returns the block number the note was created in.
+    pub fn block_num(&self) -> u32 {
+        self.block_num
+    }
+
+    /// Returns the index of the note in the note Merkle tree of the block the note was created in.
+    pub fn node_index(&self) -> u64 {
+        self.node_index_in_block as u64
+    }
```

Comment:
I would add to the doc comment a bit more info about the returned value. Specifically, we should say what the height of the tree is and what the maximum value is.

Once we reduce the height of the tree to 16, this will be less ambiguous as we can simply return `u16` or `LeafIndex<16>` (this will be even cleaner, I think).